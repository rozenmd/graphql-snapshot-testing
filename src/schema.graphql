schema {
  query: Query
  # mutation: Mutation
}
type Query {
  viewer: Viewer
}
type Viewer {
  name: String
  countries: [Country]
}
type Country {
  id: ID!
  label: String
  population: Int
}

# type Mutation {
# }
