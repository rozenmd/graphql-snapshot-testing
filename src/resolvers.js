export const getCountries = (parent, args, ctx) => {
  return [
    { id: 1, name: 'United States of America', population: 325700000 },
    { id: 61, name: 'Australia', population: 24600000 },
  ]
}

export default {
  Query: {
    // not much needed here as GraphQL can infer what a viewer should return
    viewer: (parent, args, ctx) => {
      return { name: 'some person' }
    },
  },
  Viewer: {
    countries: getCountries,
  },
  // Mutation: {

  // },
}
