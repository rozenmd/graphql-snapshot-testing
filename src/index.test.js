import graphqlHTTP from 'express-graphql'
import { getCountries } from './resolvers'
import schema from './schema'

test('the snapshot matches', async () => {
  // create a graphlhttp middleware
  const middleware = graphqlHTTP({
    //some schema containing the fields we're about to query
    schema,
    rootValue: {
      id: 'viewer',
      loaders: {
        //mock the data somehow
        countries: getCountries,
      },
    },
  })

  // create a mocked request
  const request = {
    method: 'POST',
    headers: {},
    //replace with your query
    body: { query: '{ viewer { countries { name } } }' },
  }

  // create a mock response, graphql middleware calls json() to set response data, so we need to mock it.
  const response = {
    setHeader: jest.fn(),
    end: jest.fn(),
    json: jest.fn(),
  }

  // call middleware function with mocked response and request
  await middleware(request, response)

  // get json's stub function arguments, this is actually a data returned by graphql middleware
  const responseData = response.json.mock.calls[0][0]
  // use jest.snapshot to snapshot test
  expect(responseData).toMatchSnapshot()
})
