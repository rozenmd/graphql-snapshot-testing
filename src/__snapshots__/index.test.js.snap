// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the snapshot matches 1`] = `
Object {
  "data": Object {
    "viewer": Object {
      "countries": Array [
        Object {
          "name": "United States of America",
        },
        Object {
          "name": "Australia",
        },
      ],
    },
  },
}
`;
